### PR TITLE
document ParentClosePolicy

### PR DESCRIPTION
**Pull request by sw-yx**
## What does this PR do?

document ParentClosePolicy on child workflows

## Notes to reviewers

this was a relevant discussion we had with @mastermanu  and seems an important API we dont document

### CODE REVIEW

File: docs/concepts/workflows.md [hunk 1]
```diff
@@ -258,6 +258,17 @@ One of the main reasons you would not want to use a Child Workflow is the lack o
 A Parent and Child Workflow can communicate only through asynchronous signals.
 If the executing logic has tight coupling between Workflows, it may simply be easier to use a single Workflow that can rely on a shared object's state.
 
+### ParentClosePolicy
+
+When creating a child Workflow, you can define a `ParentClosePolicy` that terminates, cancels, or abandons the Workflow if the child's parent stops execution:
```
Comment (sw-yx) [inline docs/concepts/workflows.md:263]:
```suggestion
When creating a child Workflow, you can define a [`ParentClosePolicy`](https://github.com/temporalio/api/blob/c1f04d0856a3ba2995e92717607f83536b5a44f5/temporal/api/enums/v1/workflow.proto#L44) that terminates, cancels, or abandons the Workflow if the child's parent stops execution:
```

File: docs/concepts/workflows.md [hunk 2]
```diff
@@ -258,6 +258,17 @@ One of the main reasons you would not want to use a Child Workflow is the lack o
 A Parent and Child Workflow can communicate only through asynchronous signals.
 If the executing logic has tight coupling between Workflows, it may simply be easier to use a single Workflow that can rely on a shared object's state.
 
+### ParentClosePolicy
+
+When creating a child Workflow, you can define a [`ParentClosePolicy`](https://github.com/temporalio/api/blob/c1f04d0856a3ba2995e92717607f83536b5a44f5/temporal/api/enums/v1/workflow.proto#L44) that terminates, cancels, or abandons the Workflow if the child's parent stops execution:
```
Comment (flossypurse) [inline docs/concepts/workflows.md:263]:
```suggestion
When creating a Child Workflow, you can define a [`ParentClosePolicy`](https://github.com/temporalio/api/blob/c1f04d0856a3ba2995e92717607f83536b5a44f5/temporal/api/enums/v1/workflow.proto#L44) that terminates, cancels, or abandons the Workflow Execution if the child's parent stops execution:
```

File: docs/go/workflows.md [hunk 3]
```diff
@@ -246,6 +246,34 @@ var result string
 we.Get(ctx, &result)
 ```
 
+### ParentClosePolicy
+
+When creating a child Workflow, you can define a `ParentClosePolicy` that terminates, cancels, or abandons the Workflow if the child's parent stops execution.
```
Comment (flossypurse) [inline docs/go/workflows.md:251]:
```suggestion
When creating a child Workflow, you can define a `ParentClosePolicy` that terminates, cancels, or abandons the Workflow Execution if the child's parent stops execution.
```

File: docs/concepts/workflows.md [hunk 4]
```diff
@@ -258,6 +258,17 @@ One of the main reasons you would not want to use a Child Workflow is the lack o
...
+
+- `ABANDON`: When the parent stops, don't do anything with the child workflow.
+- `TERMINATE`: When the parent stops, terminate the child workflow
+- `REQUEST_CANCEL`: When the parent stops, terminate the child workflow
```
Comment (flossypurse) [inline docs/concepts/workflows.md:267]:
```suggestion
- `ABANDON`: When the parent stops, don't do anything with the Child Workflow.
- `TERMINATE`: When the parent stops, terminate the Child Workflow
- `REQUEST_CANCEL`: When the parent stops, terminate the Child Workflow
```

File: docs/go/workflows.md [hunk 5]
```diff
@@ -246,6 +246,34 @@ var result string
...
+
+- `ABANDON`: When the parent stops, don't do anything with the child workflow.
+- `TERMINATE`: When the parent stops, terminate the child workflow
+- `REQUEST_CANCEL`: When the parent stops, terminate the child workflow
```
Comment (flossypurse) [inline docs/go/workflows.md:255]:
```suggestion
- `ABANDON`: When the parent stops, don't do anything with the Child Workflow.
- `TERMINATE`: When the parent stops, terminate the Child Workflow
- `REQUEST_CANCEL`: When the parent stops, terminate the Child Workflow
```

File: docs/go/workflows.md [hunk 6]
```diff
@@ -246,6 +246,34 @@ var result string
...
+
+You can set policies per child, which means you can opt out of propagating terminates / cancels on a per-child basis.
+This is useful for starting Child Workflows asynchronously:
+
+1. Set `ChildWorkflowOptions.ParentClosePolicy` to `ABANDON` when creating a child workflow stub.
+2. Start child workflow asynchronously using `ExecuteChildWorkflow`.
```
Comment (flossypurse) [inline docs/go/workflows.md:261]:
```suggestion
2. Start the Child Workflow Execution asynchronously using `ExecuteChildWorkflow`.
```

File: docs/go/workflows.md [hunk 7]
```diff
@@ -246,6 +246,34 @@ var result string
...
+3. Call `GetChildWorkflowExecution` on the `ChildWorkflowFuture` returned by the `ChildWorkflowFuture`
+4. Wait for the `ChildWorkflowFuture`. This indicates that the child successfully started (or start failed).
```
Comment (flossypurse) [inline docs/go/workflows.md:263]:
```suggestion
4. Wait for the `ChildWorkflowFuture`.
    This indicates that the child successfully started (or start failed).
```

File: docs/go/workflows.md [hunk 8]
```diff
@@ -246,6 +246,34 @@ var result string
...
+5. Complete parent workflow asynchronously.
```
Comment (flossypurse) [inline docs/go/workflows.md:264]:
```suggestion
5. Complete Parent Workflow Execution asynchronously.
```

File: docs/go/workflows.md [hunk 9]
```diff
@@ -246,6 +246,34 @@ var result string
...
+
+Steps 3 and 4 are needed to ensure that child workflow starts before the parent closes. If the parent initiates child workflow and immediately completes the child would never start.
```
Comment (flossypurse) [inline docs/go/workflows.md:266]:
```suggestion
Steps 3 and 4 are needed to ensure that a Child Workflow Execution starts before the parent closes. 
    If the parent initiates a Child Workflow Execution and then immediately completes, the child would never execute.
```

File: docs/java/workflows.md [hunk 10]
```diff
@@ -311,6 +311,39 @@ public class GreetingWorkflowImpl implements GreetingWorkflow {
 }
 ```
 
+### ParentClosePolicy
+
+When creating a child Workflow, you can define a `ParentClosePolicy` that terminates, cancels, or abandons the Workflow if the child's parent stops execution.
```
Comment (flossypurse) [inline docs/java/workflows.md:316]: The rest of this appears to be a copy & past from the top except with a Java example.
Same suggestions for this as well.

File: docs/go/workflows.md [hunk 11]
```diff
@@ -246,6 +246,34 @@ var result string
...
+When creating a child Workflow, you can define a `ParentClosePolicy` that terminates, cancels, or abandons the Workflow if the child's parent stops execution.
+
+- `ABANDON`: When the parent stops, don't do anything with the child workflow.
+- `TERMINATE`: When the parent stops, terminate the child workflow
+- `REQUEST_CANCEL`: When the parent stops, terminate the child workflow
+
+You can set policies per child, which means you can opt out of propagating terminates / cancels on a per-child basis.
+This is useful for starting Child Workflows asynchronously:
+
+1. Set `ChildWorkflowOptions.ParentClosePolicy` to `ABANDON` when creating a child workflow stub.
```
[the remark this answers is not in the document]
Comment (sw-yx) [reply docs/go/workflows.md:260]: no - oversight on my part

**Review by flossypurse — APPROVED** [review]

**Pull request merged**